modules/dmrpp_module/build_dmrpp_util.cc

```diff
@@ -1117,109 +1226,72 @@ void get_chunks_for_all_variables(hid_t file, D4Group *group) {
+        VERBOSE(cerr << prolog);
+        VERBOSE(btp->print_decl(cerr,"",false,false,false) );
+        VERBOSE(cerr << endl);
+
+        auto dataset  = get_h5_dataset_id(file, btp, nc4_non_coord_candidate);
+        if(dataset > 0) {
+            // If we have a valid dataset then we have a variable with data. I think.
+            // If it's not valid we skip it, I think because the associated BaseType,
+            // btp, may be a semantic object, like a dimension,  with no data
+            // associated with it.
+            try {
+                string msg;
+                if (is_unsupported_type(dataset, btp, msg)) {
+                    throw UnsupportedTypeException(msg);
                 }
-            }
-        }
 
-        try {
-            string msg;
-            if(is_unsupported_type(dataset, btp, msg)){
-                // @TODO What should really happen in this case?
-                //   - Throw an exception?
-                //   - Elide the variable from the dmr++?
-                //   - Demote dataset/var to Attribute?
-                //   - Mark the variable as "unsupported" so that it's metadata are transmitted but
-                //     it's data cannot be read.
-                throw UnsupportedTypeException(msg);
-            }
+                if (!process_variable_length_string_scalar(dataset, btp)) {
 
-            VERBOSE(cerr << prolog << "Building chunks for: " << get_type_decl(btp) << endl);
-            get_variable_chunk_info(dataset, btp);
+                    VERBOSE(cerr << prolog << "Building chunks for: " << get_type_decl(btp) << endl);
+                    get_variable_chunk_info(dataset, btp);
 
-            VERBOSE(cerr << prolog << "Annotating String Arrays as needed for: " << get_type_decl(btp) << endl);
-            add_string_array_info(dataset, btp);
-            H5Dclose(dataset);
-        }
-        catch (UnsupportedTypeException &uste){
-            // TODO - If we are going to elide a variable because it is an unsupported type, I think
-            //  that this would be the place to do it.
-            cerr << prolog << "Caught UnsupportedTypeException for variable " << btp->FQN() << " message: " << uste.what() << endl;
-            throw;
+                    VERBOSE(cerr << prolog << "Annotating String Arrays as needed for: " << get_type_decl(btp) << endl);
+                    add_string_array_info(dataset, btp);
+                }
+                H5Dclose(dataset);
+            }
+            catch (...) {
+                H5Dclose(dataset);
+                throw;
+            }
         }
-        catch (...) {
-            H5Dclose(dataset);
-            throw;
+        else {
+            VERBOSE(cerr << prolog << "Unable to open " << btp->FQN()
+            << " with the hdf5 api. Skipping chunk production. "
+            << "Everything will be ok." << endl);
         }
     }
 
     // all groups in the group
+    // for(auto g:group->groups())
```

Comment:
One thing that sonar scan complains about is commented out code. Maybe remove this?

Reply:
done